lib/api/include/irods/replica_truncate.h

```diff
@@ -0,0 +1,82 @@
+#ifndef IRODS_REPLICA_TRUNCATE_H
+#define IRODS_REPLICA_TRUNCATE_H
+
+struct RcComm;
+struct DataObjInp;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Truncate a replica for the specified data object to the specified size.
+///
+/// \parblock
+/// This API selects a replica to truncate according to the rules of POSIX truncate(2). The caller may provide keywords
+/// via condInput in order to influence the hierarchy resolution for selecting a replica to truncate.
+/// \endparblock
+///
+/// \param[in] _comm A pointer to a RcComm.
+/// \param[in] _inp \parblock
+/// DataObjInp structure which requires the following inputs:
+///     objPath - The full logical path to the target data object.
+///     dataSize - The desired size of the replica after truncating.
+///
+/// The condInput supports the following keywords for hierarchy resolution for a "write" operation:
+///     replNum - The replica number of the replica to truncate.
+///     rescName - The name of the resource with the replica to truncate. Must be a root resource.
+///     defRescName - The default resource to target in the absence of any other inputs or policy.
+///     resc_hier - Full resource hierarchy to the replica to truncate. Use with caution.
+///     irodsAdmin - Flag indicating that the operation is to be performed with elevated privileges. No value required.
```

Comment:
I went back and forth on this and landed on the actual strings for the reason you stated. I think it would be most useful for developers of new libraries to know what an API can use. I could be wrong on this, but I think most developers outside of our team are not working with C/C++, so they will need to know the actual strings.

I don't feel very strongly about this, so I'm fine to put in the C macro names instead.

Reply:
How do you feel about including both?

C devs should use the macros. For other languages, use the strings. Given we never change existing macros/string, that seems to be a decent approach.

Is there another way to handle this?
This is a C API, do we have to cover other languages?

Perhaps we need to investigate how other projects deal with this.

Reply:
I think including both is fine, sure.

As for the comment about this being a C API, I take the point, but this is basically the only documentation that will exist for calling this API (unless we have some other way of doing this of which I am unaware), so I think it wouldn't hurt to be accommodating. Thoughts?

Reply:
Well, let's consider someone attempting to use Jargon. I can't imagine users of Jargon reaching for the C documentation or even thinking to look at it since one would have to assume Jargon provides all the documentation needed for using it. Not saying it's impossible, but probably unlikely. Communicating usage of the Jargon APIs is on the library developer. The same thing applies to the PRC.

I'm a bit hesitant about mentioning the strings directly because some C devs may choose to use them when they should be using the macros _(because it's less error-prone)_.

If devs want to know what the macros map to, they can look it up in the header file where the keywords are defined.

Reply:
Seems fine to me. I will replace the literal keywords with the macro names.

Reply:
Done